### PR TITLE
Don't override windows blue when in xterm mode

### LINKED ISSUE #59: (Windows) Consider using standard color scheme when term is xterm like
perhaps check for:

`process.platform === 'win32' && !process.env.TERM`
or
`process.platform === 'win32' && !(/xterm/).test(process.env.TERM || "")`

I am suggesting this because I use node through a solarized color shell (admittedly through cygwin ssh).

Thanks.

### PR DESCRIPTION
**Pull request by weikinhuang**
fixes #59

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -12,7 +13,7 @@ function Chalk(options) {
 }
 
 // use bright blue on Windows as the normal blue color is illegible
-if (process.platform === 'win32') {
+if (process.platform === 'win32' && !xtermRegexp.test(process.env.TERM || 'dumb')) {
```
Comment (sindresorhus) [inline index.js]:
```js
if (process.platform === 'win32' && !/^xterm/i.test(process.env.TERM)) {
```

Comment (jbnicolai) [inline index.js]: Fully agree that there's no need to declare the regex. This is not a hot code path (very much not so)

Comment (weikinhuang) [inline index.js]: ok, i'll make that change then

**Comment by jbnicolai** [comment]
@weikinhuang appreciate the effort, and happy to merge, but would you be willing to add a (small) test asserting the correct behaviour?

**Comment by weikinhuang** [comment]
@jbnicolai but the test will only work in windows

File: index.js [hunk 2]
```diff
@@ -12,7 +12,7 @@ function Chalk(options) {
 }
 
 // use bright blue on Windows as the normal blue color is illegible
-if (process.platform === 'win32') {
+if (process.platform === 'win32' && !(/^xterm/i).test(process.env.TERM || 'dumb')) {
```
Comment (sindresorhus) [inline index.js]: Please copy paste what I commented, the regex doesn't need parens and `dumb` check is invalid.

**Comment by jbnicolai** [comment]
@weikinhuang `process.env` can be mocked. See https://github.com/sindresorhus/supports-color/blob/master/test.js for an example of how to test emulating different environments.

**Comment by jbnicolai** [comment]
@weikinhuang and last but not least: it'd be nice if you'd squash the pr into a single commit, although I'd be happy to do this for you.

**Comment by weikinhuang** [comment]
@jbnicolai Will do, I like keeping separate commits right up until the point of completion, then squashing afterwards usually. Thanks for the comments!

File: index.js (hunk 2 above)
Comment (weikinhuang) [inline index.js]: will do.

**Comment by weikinhuang** [comment]
@jbnicolai Actually I am unable to mock `process.env` because that check is not a run time override. Would any of you be able to provide some tips for mocking that?

**Comment by weikinhuang** [comment]
Nevermind, figured it out, had to use `require-uncached`.

**Comment by jbnicolai** [comment]
Precisely :)

**Comment by weikinhuang** [comment]
Updated + tests

**Comment by weikinhuang** [comment]
However I think this would fail since I can't mock `process.platform` because it's a read only property. Should I just run the test when the platform is `win32`?

**Comment by sindresorhus** [comment]
You can mock it with `Object.defineProperty(process, 'platform', {value: 'win32'});`.

**Comment by weikinhuang** [comment]
Cool, I always forget about `Object.defineProperty`

**Comment by weikinhuang** [comment]
@sindresorhus do you know why the test is failing on `v0.10`?

**Comment by sindresorhus** [comment]
I think it has something to do with us overriding `process.platform` and module path resolution logic on Node 0.10. I think we can just `if` check if it's node 0.12 or newer for those tests.

**Comment by weikinhuang** [comment]
Updated

File: test.js [hunk 3]
```diff
@@ -1,6 +1,9 @@
 'use strict';
 var assert = require('assert');
 var chalk = require('./');
+var requireUncached = require('require-uncached');
+var resolveFrom = require('resolve-from');
+var semver = require('semver');
```
Comment (kevva) [inline test.js]: Can you move these up before `var chalk = require('./');` and after `var assert = require('assert');`?

Comment (weikinhuang) [inline test.js]: sure

**Comment by weikinhuang** [comment]
Updated

**Comment by sindresorhus** [comment]
Looks good. Thanks :)

**Comment by sindresorhus** [comment]
Will do a new release when https://github.com/sindresorhus/chalk/issues/58 is resolved. In the meantime you can use it with `npm install --save sindresorhus/chalk` ;)

**Pull request closed without merging**